node/index.d.ts

```diff
@@ -1891,8 +1891,27 @@ declare module "dns" {
         priority: number
     }
 
-    export function lookup(domain: string, family: number, callback: (err: Error, address: string, family: number) => void): string;
-    export function lookup(domain: string, callback: (err: Error, address: string, family: number) => void): string;
+    // Supported getaddrinfo flags
+    export const ADDRCONFIG: number;
+    export const V4MAPPED: number;
+
+    export interface LookupOptions {
```

Comment:
in case typescript 2.2 features are already allowed to be used we should use object as base for LookupOptions as current definition of LookupOptions allows anything (e.a. string,...).